test/ClangImporter/const_values.swift

```diff
@@ -0,0 +1,95 @@
+// RUN: %empty-directory(%t/src)
+// RUN: split-file %s %t/src
+
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck %t/src/main.swift \
+// RUN:   -import-bridging-header %t/src/test.h \
+// RUN:   -module-name main -I %t -emit-sil | %FileCheck %s
+
+//--- test.h
+#include <stdbool.h>
+
+#define MACRO_INT 42
+
+const int const_int = 42;
+static int static_int = 42;
+static const int static_const_int = 42;
+
+static const bool static_const_bool = true;
+static const char static_const_char = 42;
+static const long static_const_long = 42;
+static const float static_const_float = 42.0;
+static const double static_const_double = 42.0;
```

Comment:
Some more edge cases it might be good to test:

* Literal uses, say, `ull` to specify a different type from the constant. (Trying to make sure we don't infer a type from the literal like we might in a macro.)
* Literal is too big for the type of the constant it's being assigned to (e.g. a `char` initialized to `500`).
* Initializer refers to another constant which clang could potentially evaluate to find the value.
* Initializer refers to another constant which clang cannot evaluate.

Reply:
Maybe not that interesting, but I think it might be nice to test some C++-y things as well:
* Constexpr globals (that potentially are computed by constexpr functions)
* Constants in a namespace (as namespaces are imported as enums)
* Constants in class scope (static and non-static), even if we currently do not import them

Reply:
👍 will add all these tests, I started gathering the list of TODOs in the PR description

Reply:
I think this should now all be covered by tests.